### PR TITLE
Ensure all references to static locals are replaced with proper field references.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems?id=264417.

@dotnet/roslyn-compiler Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
CC @MattGertz For ask mode approval. This is a fix for a "Watson" crash.

**Comment by MattGertz** [comment]
Approved pending the usual.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Source/StaticLocals.vb [hunk 1]
```diff
@@ -922,5 +922,172 @@ End Module
             CompileAndVerify(compilation)
         End Sub
 
+        <Fact>
+        <WorkItem(264417, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=264417")>
+        Public Sub InitializeWithAsNew_01()
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Source/StaticLocals.vb]: Consider naming the tests `InitializeStructureWithAsNew_` and `InitializeClassWithAsNew_` so it's clear what the difference is between the pairs of tests.

File: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Source/StaticLocals.vb [hunk 2]
```diff
@@ -922,5 +922,172 @@ End Module
...
+]]>)
+        End Sub
+
+        <Fact>
+        <WorkItem(264417, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=264417")>
+        Public Sub InitializeWithAsNew_03()
+            Dim compilationDef =
+<compilation>
+    <file name="a.vb"><![CDATA[
+Public Class S1
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Source/StaticLocals.vb]: Consider naming `C1` or `C` here and in `_04`.

File: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Source/StaticLocals.vb (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Source/StaticLocals.vb]: I don't think this specialization is necessary.

**Comment by cston** [comment]
LGTM

**Comment by VSadov** [comment]
LGTM

**Pull request merged**